crates/lsp/src/lib.rs

```diff
@@ -133,11 +155,44 @@ impl<L: LSPLang> LanguageServer for Backend<L> {
       .await;
   }
 
-  async fn did_change_watched_files(&self, _: DidChangeWatchedFilesParams) {
+  async fn did_change_watched_files(&self, params: DidChangeWatchedFilesParams) {
     self
       .client
       .log_message(MessageType::INFO, "watched files have changed!")
       .await;
+    
+    // Check if any of the changed files are configuration files
+    let mut config_changed = false;
+    for change in &params.changes {
```

Comment:
just reload here, the file watcher requesst already guarantee that only yml is watched